core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedService.java

```diff
@@ -398,7 +402,7 @@ private HttpResponse convertResponse(ServiceRequestContext ctx, @Nullable Object
 
         if (result instanceof HttpResult) {
```

Comment:
note; Perhaps in the future we may just remove this block since `CompositeResponseConverterFunction` has this logic already.

I understand this can't be removed this iteration due to breaking changes though

Reply:
I've tried that but it might fail depends on the implementation:
https://github.com/line/armeria/blob/d34cee70d383c16cb394145a73b54ca66d126885/kotlin/src/test/kotlin/com/linecorp/armeria/server/kotlin/SuspendingAnnotatedServiceTest.kt#L374-L377

Reply:
Thanks, I also checked that the content unwrapping logic is the reason for the failure
```
result = httpResult.content();
```